test/corpus/statements.txt

```diff
@@ -899,3 +899,220 @@ end
           (tuple_type
             (constant)
             (constant)))))))
+
+================
+lib external var
+================
+
+lib C
+  $errno : Int32
+end
+
+---
+
+(source_file
+  (lib_def
+    (constant)
+    (global_var
+      (identifier)
+      (constant))))
+
+=========
+empty lib
+=========
+
+lib C
+end
+
+---
+
+(source_file
+  (lib_def
+    (constant)))
+
+=============
+top-level fun
+=============
+
+fun my_fun_name(param1 : Type, param2 : Type) : ReturnType
+  puts "hello"
+end
```

Comment:
As another edge case test:
```
fun
extreme_fun
(
param1
:
Type
,
param2
:
Type
)
:
ReturnType
  puts "hello"
end
```
should have the same syntax tree.

The parser doesn't handle this at the moment, so you could add as a TODO.